scalding-core/src/test/scala/com/twitter/scalding/ExecutionTest.scala

```diff
@@ -417,5 +415,15 @@ class ExecutionTest extends WordSpec with Matchers {
 
       assert(result.waitFor(Config.default, Local(true)).get == 0.to(10).reverse)
     }
+
+    "can hashCode, compare, and run a long sequence" in {
+      val execution = Execution.sequence((1 to 100000).toList.map(Execution.from(_)))
```

Comment:
It won't solve the problem, and it is indeed what I meant for the last part.  We (still) see the latter error if we (only) override hashCode to be eager.